### ISSUE TITLE
Bundler 2.5.20 appears to be broken (missing dependencies after bundle install)

### ISSUE THREAD

**Issue by rgaufman** [rubygems/rubygems#8078]
### Describe the problem as clearly as you can

Despite bundle install completing successfully, running bin/rails c complains that dependencies are missing:

```
$ bundle install
Bundle complete! 102 Gemfile dependencies, 183 gems now installed.
Gems in the groups 'development' and 'test' were not installed.
Bundled gems are installed into `./vendor/bundle`
2 installed gems you directly depend on are looking for funding.
  Run `bundle fund` for details
$ bin/rails c
Could not find rtp-0.1.6 in https://github.com/tetherit/rtp.git (at /data/deployer/tetherbox/vendor/cache/rtp-32c1d95f87de@32c1d95) nor rtsp-0.4.5 in https://github.com/rgaufman/rtsp.git (at /data/deployer/tetherbox/vendor/cache/rtsp-b3e04e5f92f3@b3e04e5) nor tetherx-mqtt-2.0.1 in git@github.com:tetherit/tetherx-mqtt.git (at /data/deployer/tetherbox/vendor/cache/tetherx-mqtt-e3572aca2b0e@e3572ac) nor tetherx-mqtt2-2.0.1 in git@github.com:tetherit/tetherx-mqtt2.git (at /data/deployer/tetherbox/vendor/cache/tetherx-mqtt2-bef360c2f18b@bef360c) nor serialport-1.3.3 in https://github.com/rgaufman/ruby-serialport.git (at /data/deployer/tetherbox/vendor/cache/ruby-serialport-7946ff4da617@7946ff4) nor ws_discovery-0.0.3 in https://github.com/rgaufman/ws-discovery.git (at /data/deployer/tetherbox/vendor/cache/ws-discovery-833f9a2fcd1d@833f9a2) nor rails_semantic_logger-4.17.0 in https://github.com/reidmorrison/rails_semantic_logger.git (at /data/deployer/tetherbox/vendor/cache/rails_semantic_logger-6e1291f49c74@6e1291f) nor semantic_logger-4.16.1 in https://github.com/reidmorrison/semantic_logger.git (at /data/deployer/tetherbox/vendor/cache/semantic_logger-9837e6b7689b@9837e6b) nor bootstrap-slider-rails-9.8.0 in https://github.com/rgaufman/bootstrap-slider-rails.git (at /data/deployer/tetherbox/vendor/cache/bootstrap-slider-rails-e2678ad7fab0@e2678ad)
```

### Did you try upgrading rubygems & bundler?

Running latest. Downgrading bundler from 2.5.20 to 2.5.16 fixes the problem.

### Which command did you run?

As mentioned above.

### What were you expecting to happen?

For bin/rails c to start after a successful bundle install.

### What actually happened?

It complains I need to run bundle, but I already ran it.

**Comment by deivid-rodriguez** [comment]
Thanks. If you could share steps to reproduce, that would be ideal.

My first hunch is that this is related to local overrides for git sources. Can you share you `bundle env`, if sharing a completely reproducible example is not possible?

**Comment by rgaufman** [comment]
I think it's related to this, closing as it's a duplicate: https://github.com/rubygems/rubygems/issues/8069

**Comment by deivid-rodriguez** [comment]
I'm not sure this is actually a duplicate, but happy to leave this closed, and once the solution for #8069 is released, you can try again and we can reopen if the issue is still there.
